web/packages/shared/utils/semVer.ts

```diff
@@ -16,31 +16,15 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-export const compareSemVers = (a: string, b: string): -1 | 1 | 0 => {
-  const splitA = a.split('.');
-  const splitB = b.split('.');
+import compare from 'semver/functions/compare';
+import major from 'semver/functions/major';
 
-  if (splitA.length < 3 || splitB.length < 3) {
-    return -1;
-  }
+// Re-export specific functions from `semver/functions/*` to avoid bundling
+// the full semver package.
 
-  const majorA = parseInt(splitA[0]);
-  const majorB = parseInt(splitB[0]);
-  if (majorA !== majorB) {
-    return majorA > majorB ? 1 : -1;
-  }
-
-  const minorA = parseInt(splitA[1]);
-  const minorB = parseInt(splitB[1]);
-  if (minorA !== minorB) {
-    return minorA > minorB ? 1 : -1;
-  }
-
-  const patchA = parseInt(splitA[2].split('-')[0]);
-  const patchB = parseInt(splitB[2].split('-')[0]);
-  if (patchA !== patchB) {
-    return patchA > patchB ? 1 : -1;
-  }
-
-  return 0;
+export {
+  compare,
```

Comment:
Now that we're using `compare` from the library, it might be worthwhile to add some tests for pre-releases. Do auto-updates work with pre-releases at all? It might not be important for `auto_update.tools_version` if auto-updates don't ever use pre-releases. But `min_client_version` for sure returns a pre-release, so I wonder if this has any impact.